src/backend/distributed/commands/multi_copy.c

```diff
@@ -162,6 +164,8 @@ struct CopyPlacementState
 	/* State of shard to which the placement belongs to. */
 	CopyShardState *shardState;
 
+	int32 groupId;
```

Comment:
`/* node group ID of the placement */`

(we should probably find a better name than "groupId" throughout the code base)

Reply:
Yeah maybe node id?

Reply:
Node ID is a different concepts. There can be multiple nodes in the same group (read replicas). So a "group" is effectively a logical database that can be replicated across one or more servers.